cdap-ui/app/cdap/components/AbstractWidget/SqlConditionsWidget/Rule.tsx

```diff
@@ -0,0 +1,173 @@
+    },
+    stageRow: {
+      display: 'grid',
+      gridTemplateColumns: '45% 45% 10%',
+      marginBottom: '5px',
+    },
+    stageName: {
+      display: 'flex',
+      alignItems: 'center',
+    },
+    fieldEqualityContainer: {
+      display: 'flex',
+      alignItems: 'center',
+    },
+    fieldEquality: {
+      color: 'white',
+      width: '20px',
+      height: '20px',
+      textAlign: 'center',
+      backgroundColor: theme.palette.grey[100],
+      marginLeft: '5px',
+    },
+    deleteButton: {
+      position: 'absolute',
+      top: '-16px',
+      right: '-16px',
+      color: theme.palette.red[200],
+    },
+    andButton: {
+      borderRadius: '0',
+      color: 'white',
+      backgroundColor: theme.palette.grey[100],
+      padding: '5px',
+      '&:hover': { textDecoration: 'none' },
+    },
+  };
+};
+
+interface IRuleProps extends WithStyles<typeof styles> {
+  rule: IRule;
+  inputSchema: IInputSchema;
+  disabled: boolean;
+  ruleIdx: number;
+  rulesCount: number;
+  addRule: () => void;
+  updateRule: (rule: IRule) => void;
+  deleteRule: (rule: number) => void;
+}
+
+const Rule: React.FC<IRuleProps> = ({
+  rule,
+  inputSchema,
+  rulesCount,
+  ruleIdx,
+  disabled,
+  addRule,
+  updateRule,
+  deleteRule,
+  classes,
+}) => {
+  const [hovered, setHovered] = useState(false);
+  const last = ruleIdx === rulesCount - 1;
+  const fieldChange = (stageIdx, event) => {
+    const newRule = rule.map((row, i) => {
+      if (i === stageIdx) {
+        return { ...row, fieldName: event.target.value };
+      }
+      return row;
+    });
+    updateRule(newRule);
+  };
+  return (
+    <div
+      className={classes.root}
+      onMouseEnter={() => setHovered(true)}
+      onMouseLeave={() => setHovered(false)}
+    >
+      <Paper className={classes.rule}>
+        <If condition={hovered && rulesCount !== 1 && !disabled}>
+          <IconButton
+            size="small"
+            className={classes.deleteButton}
+            onClick={() => deleteRule(ruleIdx)}
+          >
+            <CancelIcon fontSize="large" />
+          </IconButton>
+        </If>
+        {rule.map((stage, i) => (
+          <div className={classes.stageRow}>
+            <div className={classes.stageName}>{stage.stageName}</div>
+            <div>
+              <Select
+                widgetProps={{
+                  values: inputSchema[stage.stageName],
+                }}
+                value={stage.fieldName}
+                onChange={(e) => fieldChange(i, e)}
+              />
+            </div>
+            <If condition={i !== rule.length - 1}>
+              <div className={classes.fieldEqualityContainer}>
+                <span className={classes.fieldEquality}>=</span>
+              </div>
+            </If>
+          </div>
+        ))}
+      </Paper>
+      <div className={classes.actionButtons}>
+        <If condition={!last}>
+          <span className={classnames('btn', 'btn-sm', classes.andButton)}>AND</span>
```

Comment:
try using material ui button

Reply:
@elfenheart should I just make this a span, remove btn styling? This is not really a button, it has no onClick action?

Reply:
that works